app/models/world_location.rb

```diff
@@ -13,20 +13,21 @@ def self.reset_cache
 
   def self.all
     cache_fetch("all") do
-      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |l|
-        new(l) if l.format == "World location" && l.details && l.details.slug.present?
+      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |response|
+        location = Location.new(response.to_hash)
+        if location.format == "World location" && location.details &&
+            location.slug.present?
+          self.new(location)
+        end
       end
       world_locations.compact
     end
   end
 
   def self.find(location_slug)
     cache_fetch("find_#{location_slug}") do
-      response = Services.worldwide_api.world_location(location_slug)
-      if response
-        data = OpenStruct.new(response.to_hash)
-      end
-      self.new(data) if data
+      response = Services.worldwide_api.world_location(location_slug)&.to_hash
```

Comment:
@pmanrubia the location sruct no longer exists.. see https://github.com/alphagov/smart-answers/blob/3c8ff7a363617d959145b70ac58a7d934757b1ec/app/models/world_location.rb#L61